components/docker-up/docker-up/main.go

```diff
@@ -161,10 +175,91 @@ func runWithinNetns() (err error) {
 	return nil
 }
 
+var allowedDockerArgs = map[string]string{
+	"remap-user": "userns-remap",
+}
+
+func setUserArgs(args []string) ([]string, error) {
+	userArgs, exists := os.LookupEnv(DaemonArgs)
+	if !exists {
+		return args, nil
+	}
+
+	var providedDockerArgs map[string]string
+	if err := json.Unmarshal([]byte(userArgs), &providedDockerArgs); err != nil {
+		return nil, xerrors.Errorf("unable to deserialize docker args: %w", err)
+	}
+
+	for userArg, userValue := range providedDockerArgs {
+		mapped, exists := allowedDockerArgs[userArg]
+		if !exists {
+			continue
+		}
+
+		if userArg == "remap-user" {
+			id, err := strconv.Atoi(userValue)
+			if err != nil {
+				return nil, err
+			}
+
+			for _, f := range []string{"/etc/subuid", "/etc/subgid"} {
+				err := adaptSubid(f, id)
+				if err != nil {
+					return nil, xerrors.Errorf("could not adapt subid files: %w", err)
+				}
+			}
+
+			args = append(args, "--"+mapped, "gitpod")
+		} else {
+			args = append(args, "--"+userArg, userValue)
+		}
+	}
+
+	return args, nil
+}
+
+func adaptSubid(oldfile string, id int) error {
+	uid, err := os.Open(oldfile)
+	if err != nil {
+		return err
+	}
+
+	newfile, err := os.Create(oldfile + ".new")
+	if err != nil {
+		return err
+	}
+
+	if id != 0 {
+		newfile.WriteString(fmt.Sprintf("gitpod:%d:%d\n", 1, id))
+		newfile.WriteString("gitpod:33333:1\n")
+
+	} else {
+		newfile.WriteString("gitpod:33333:1\n")
+		newfile.WriteString(fmt.Sprintf("gitpod:%d:%d\n", 1, 33332))
+		newfile.WriteString(fmt.Sprintf("gitpod:%d:%d\n", 33334, 32200))
+	}
```

Comment:
How about this? I couldn't get the point what `32200` mean. What does it mean?
```suggestion
	gitpodUserId := 33333
	mappingFmt := func(username string, id int, size int) string { return fmt.Sprintf("%s:%d:%d\n", username, id, size) }

	if id != 0 {
		newfile.WriteString(mappingFmt("gitpod", 1, id))
		newfile.WriteString(mappingFmt("gitpod", gitpodUserId, 1))
	} else {
		newfile.WriteString(mappingFmt("gitpod", gitpodUserId, 1))
		newfile.WriteString(mappingFmt("gitpod", 1, gitpodUserId-1))
		newfile.WriteString(mappingFmt("gitpod", gitpodUserId+1, 33200))
	}
```